src/ray/core_worker/reference_count.cc

```diff
@@ -103,20 +195,79 @@ void ReferenceCounter::DeleteReferences(const std::vector<ObjectID> &object_ids)
     if (it == object_id_refs_.end()) {
       return;
     }
+    it->second.local_ref_count = 0;
+    it->second.submitted_task_ref_count = 0;
+    if (distributed_ref_counting_enabled_ && !it->second.CanDelete()) {
+      RAY_LOG(ERROR)
+          << "ray.internal.free does not currently work for objects that are still in "
```

Comment:
I think we should just not mess with the references in `ray.internal.free` and just remove from plasma instead. Thoughts @ericl?

Reply:
Oh yeah, that makes sense, it interacts poorly with the ref counting. I just put this here to make the current unit tests pass, but we can revisit.